### ISSUE TITLE
Failure to detect `noqa` statement

### ISSUE THREAD

**Issue by ddelange** [jsh9/pydoclint#109]
From https://github.com/jsh9/pydoclint/issues/108:
```py
import typing as tt

def read_jsonl(
    path_jsonl: str,
) -> tt.Iterator[dict[int, tuple[int, int, int]]]:
    """Read a jsonl file, one record at a time.

    Args:
        path_jsonl: Path to a jsonl file.

    Yields:
        One record.
    """
    with Path(path_jsonl).open("rb") as fp:
        yield from fp
```

Will flag false positives:
```
    29: DOC201: Function `mock_loop_over_video` does not have a return section in docstring
    29: DOC403: Function `mock_loop_over_video` has a "Yields" section in the docstring, but there are no "yield" statements, or the return annotation is not a Generator/Iterator/Iterable.
```

I tried to silence them with a noqa:

```py
import typing as tt

def read_jsonl(  # noqa: DOC201, DOC403, RUF100 false positives ref https://github.com/jsh9/pydoclint/issues/108
    path_jsonl: str,
) -> tt.Iterator[dict[int, tuple[int, int, int]]]:
    """Read a jsonl file, one record at a time.

    Args:
        path_jsonl: Path to a jsonl file.

    Yields:
        One record.
    """
    with Path(path_jsonl).open("rb") as fp:
        yield from fp
```

But that still flags the same false positives.

Using pre-commit hook:
```yaml
- repo: https://github.com/jsh9/pydoclint
  rev: 0.3.8
  hooks:
  - id: pydoclint
```

With pyproject.toml:
```toml
[tool.pydoclint]
style = "google"
arg-type-hints-in-docstring = false
check-return-types = false
check-yield-types = false
```

**Comment by jsh9** [comment]
Hi @ddelange , if you try to use `from typing import Iterator`, does the issue go away?

**Comment by ddelange** [comment]
Yes, when I change the import, the noqa's would be no longer necessary. This issue is about the noqa statement having no effect.

**Comment by jsh9** [comment]
I see. Let me look into it in the next few days then. Thanks!

**Comment by ddelange** [comment]
No hurries, have a nice weekend!

**Comment by jsh9** [comment]
Hi @ddelange : sorry for the late reply!

To make `noqa` work, you'd need to use `flake8` rather than directly using `pydoclint`.

Here is the documentation ([link](https://github.com/jsh9/pydoclint?tab=readme-ov-file#22-as-a-flake8-plugin)):

> 2.2. As a flake8 plugin
> Once you install pydoclint you will have also installed flake8. Then you can run:
> 
> flake8 --select=DOC <FILE_OR_FOLDER>
>
> If you don't include --select=DOC in your command, flake8 will also run other built-in flake8 linters on your code.

I verified it with your code example, and I was able to get 0 violations.
